ambry-frontend/src/test/java/com.github.ambry.frontend/FrontendIntegrationTest.java

```diff
@@ -354,6 +385,30 @@ private void getBlobAndVerify(String blobId, HttpHeaders expectedHeaders, ByteBu
   }
 
   /**
+   * Gets the blob with blob ID {@code blobId} and verifies that the blob is not returned as blob is not modified
+   * @param blobId the blob ID of the blob to GET.
+   * @throws Exception
+   */
+  private void getNotModifiedBlobAndVerify(String blobId)
+      throws Exception {
+    HttpHeaders headers = new DefaultHttpHeaders();
+    SimpleDateFormat dateFormat = new SimpleDateFormat(RestUtils.HTTP_DATE_FORMAT, Locale.US);
```

Comment:
I did try that intially. But over the network (IntegrationTest) will send as String and not as Date. if I set it as Date in this test case, restRequest.getArgs().getHeader() will return an Object which will be Date() object and not string. So we don't actually mimic real system. And thats why.